### ISSUE TITLE
leoG.findUNL() always returns None when the UNL does not contain any "-->", that is, it cannot find any root node

### ISSUE THREAD

**Issue by SegundoBob** [leo-editor/leo-editor#2455]
I do not know whether or not this is a bug.  That is, I don't know if any objectionable Leo-Editor behavior results.

I think it is unlikely that this is the intended behavior.  I consider it objectionable edge-case behavior, but I didn't write it and I don't maintain it.

I found 11 calls to leoG.findUNL() in leoPyRef.leo.  These are 11 places where objectionable edge-case behavior could occur.

I've used UNL's extensively for years.  Thanks for cleaning up their support.  I hope you choose to fix this edge case so that I can retire my code for finding the position specified by a UNL.

The calling sequence for leoG.findUNL() is strange.  It would be less strange if its first argument was simply a UNL instead of unl.split('-->').  Why not do the split in leoG.findUNL(), instead of at every call to it?

I'm also surprised that findUNL() is a leoG method and not a commander method.  I would expect a calling sequence as follows:

cmdr.findUNL(unl)

instead of

leoG.findUNL(unl.split('-->'), cmdr)

--------
Version:
Leo 6.6b2-devel, devel branch, build 0ce2fa9ad5
2022-02-24 09:55:29 -0600
Python 3.8.10, PyQt version 5.12.8
linux
---------

SegundoBob

**Comment by tbpassin** [comment]
I wondered about it being a g.*** instead of a c.*** method too.  But it might navigate you to a different outline, so I think it makes some sense because it doesn't only involve the current c.

**Comment by edreamleo** [comment]
@SegundoBob

A quick test shows that control-clicking a UNL without any "-->" does work.

Yes, it would be more logical to do the split inside g.findUNL, and it would be reasonable to make findUNL a commander method. However, all such changes could break existing scripts, so I am inclined not to make any such changes.

Hmm. g.findUNL could test the type of the unlList1 arg, and do the split if the "list" is actually a string. But I would rather leave things as they are :-)

**Comment by SegundoBob** [comment]
@edreamleo wrote "A quick test shows that control-clicking a UNL without any "-->" does work."

 I'm glad to hear this.  I never claimed your test above would not work.  I claimed and I still claim that leoG.findUNL() returns None when given a valid UNL for a root node.

This is shown by executing Ctrl+B on the only node in the Leo-Editor  file in the zip file attached below:

[findUNL.zip](https://github.com/leo-editor/leo-editor/files/8165379/findUNL.zip)

This Ctrl+B will execute the followintg:


```
@language python

unl = p.get_UNL()
g.es(unl)
g.es(g.findUNL(unl.split('-->'), c))
```
Regards,
SegundoBob

**Comment by tbpassin** [comment]
@SegundoBob,  I didn't mean to sound like I was doubting what you said.  I only did a few tests to see what the boundaries of the problem are.

After more testing, I have found that `g.handleUnl()` navigates correctly when there is a trailing hash character and no '-->'.  My earlier testing was done using CTRL-clicks.  And calling `g.findUNL()` agrees with what you have reported.

So `g.findUNL()` and `g.handleUnl()` work differently.  I had assumed they understand UNLs the same way - except for doing the actual navigation - and I'm clear that they should.

And I consider the current behavior of `g.findUNL()` buggy because as @SegundoBob said, it can never find a root node.
